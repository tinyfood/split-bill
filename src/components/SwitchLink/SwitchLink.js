import React from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import { EditPageIcon } from '../../styles/icons'

const SwitchLink = withRouter(props => (
  <NavLink
    to={props.location.pathname === '/' ? '/result' : '/'}
    style={{
      display: 'block',
      width: '96%',
      padding: '10px 0',
      margin: '10px auto',
      borderRadius: '4px',
      backgroundColor: '#2ecc71',
      textTransform: 'uppercase',
      textAlign: 'center',
      textDecoration: 'none',
      color: '#ffffff',
      transition: 'width .3s ease-out'
    }}
    activeStyle={{
      margin: '58px auto -60px',
      width: '80px',
      height: '80px',
      borderTopLeftRadius: '160px',
      borderTopRightRadius: '160px'
    }}
  >
    {props.location.pathname === '/' ? 'Split Bill' : <EditPageIcon />}
  </NavLink>
))

export default SwitchLink
