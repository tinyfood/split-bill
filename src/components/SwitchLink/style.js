export const defaultStyle = {
  display: 'block',
  width: '96%',
  padding: '10px 0',
  margin: '10px auto',
  borderRadius: '4px',
  backgroundColor: '#2ecc71',
  textTransform: 'uppercase',
  textAlign: 'center',
  textDecoration: 'none',
  color: '#ffffff',
  transition: 'width .3s ease-out'
}

export const activeStyle = {
  margin: '58px auto -60px',
  width: '80px',
  height: '80px',
  borderTopLeftRadius: '160px',
  borderTopRightRadius: '160px'
}
