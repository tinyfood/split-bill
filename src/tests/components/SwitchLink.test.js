import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import SwitchLink from '../../components/SwitchLink/SwitchLink'

test('should render SwitchLink correctly', () => {
  const links = {
    firstLink: '/',
    secondLimk: 'blog'
  }

  const wrapper = shallow(<SwitchLink {...links} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
