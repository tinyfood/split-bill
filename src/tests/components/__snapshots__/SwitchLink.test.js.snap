// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SwitchLink correctly 1`] = `
<Route>
  <Component />
</Route>
`;
